util/entropy.c

```diff
@@ -16,61 +18,157 @@
  * XXX win32 where there is no concept of /dev/urandom.
  */
 
+struct entropy_read
+{
+	int fd;
+};
+
 /**
- * entropy_read(buf, buflen):
- * Fill the given buffer with random bytes provided by the operating system.
+ * entropy_read_open():
+ * Initialize the ability to produce random bytes, and return a cookie.
```

Comment:
The words `operating system` are important here.  I don't want to lose the distinction between "ask the OS for entropy" and "get entropy from our internal entropy pool" (in `crypto_entropy`).